providers/edge3/src/airflow/providers/edge3/worker_api/routes/ui.py

```diff
@@ -100,3 +105,57 @@ def jobs(
         jobs=result,
         total_entries=len(result),
     )
+
+
+class MaintenanceRequest(BaseModel):
+    """Request body for maintenance operations."""
+
+    maintenance_comment: Annotated[str, Field(description="Comment describing the maintenance reason.")]
+
+
+@ui_router.post(
+    "/worker/{worker_name}/maintenance",
+)
+def request_worker_maintenance(
+    worker_name: str,
+    maintenance_request: MaintenanceRequest,
+    session: SessionDep,
+) -> None:
+    """Put a worker into maintenance mode."""
+    # Check if worker exists first
+    worker_query = select(EdgeWorkerModel).where(EdgeWorkerModel.worker_name == worker_name)
+    worker = session.scalar(worker_query)
+    if not worker:
+        raise HTTPException(status_code=404, detail=f"Worker {worker_name} not found")
+
+    # Format the comment with timestamp and username (username will be added by plugin layer)
+    formatted_comment = f"[{datetime.now().strftime('%Y-%m-%d %H:%M')}] - UI user put node into maintenance mode\nComment: {maintenance_request.maintenance_comment}"
```

Comment:
Can you fetch the user name from the `depends` (or add a user depends as FastAPI decorator) and make this into the formatted string such that it is directly visible who made it to maintenance?

Reply:
Replication of commit https://github.com/apache/airflow/commit/ec77bd65b96fd27eb601aeda51d4fb92354bc301 (4)
```suggestion
    formatted_comment = f"[{datetime.now().strftime('%Y-%m-%d %H:%M')}] - {user.get_name()} put node into maintenance mode\nComment: {maintenance_request.maintenance_comment}"
```